src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -80,9 +80,10 @@ private TimeZoneInfo(byte[] data, string id, bool dstDisabled)
             }
             _displayName = _standardDisplayName;
 
-            GetDisplayName(Interop.Globalization.TimeZoneDisplayNameType.Generic, ref _displayName);
-            GetDisplayName(Interop.Globalization.TimeZoneDisplayNameType.Standard, ref _standardDisplayName);
-            GetDisplayName(Interop.Globalization.TimeZoneDisplayNameType.DaylightSavings, ref _daylightDisplayName);
+            string uiCulture = CultureInfo.CurrentUICulture.Name.Length == 0 ? "en-US" : CultureInfo.CurrentUICulture.Name; // ICU doesn't work nicely with Invariant
```

Comment:
Maybe make `en-US` a `private const` and call it `FallbackCultureName` or similar?